packages/colorpicker/src/elements/ColorDialog/index.tsx

```diff
@@ -0,0 +1,107 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import React, { useState, useRef, forwardRef, HTMLAttributes } from 'react';
+import PropTypes from 'prop-types';
+import mergeRefs from 'react-merge-refs';
+import { GARDEN_PLACEMENT } from '@zendeskgarden/react-modals';
+import Chevron from '@zendeskgarden/svg-icons/src/16/chevron-down-stroke.svg';
+import { ColorPicker, IColorPickerProps } from '../ColorPicker';
+import { IRGBColor, IColorPickerState } from '../ColorPicker/reducer';
+import {
+  StyledIcon,
+  StyledButton,
+  StyledFauxInput,
+  StyledDialogPreview,
+  StyledTooltipModal,
+  StyledTooltipBody
+} from '../../styled';
+
+export interface IColorDialogProps extends IColorPickerProps {
+  /** A callback function that is called when the color dialog is closed */
+  onClose?: (selectedColor: string | IRGBColor | IColorPickerState) => void;
+  /** Positions the placement of the color dialog */
+  placement?: GARDEN_PLACEMENT;
+}
+
+/**
+ * @extends HTMLAttributes<HTMLButtonElement>
+ */
+export const ColorDialog = forwardRef<
```

Comment:
I think the `ColorDialog` API will need to be fundamentally refactored to off flexibility similar to [Select](https://garden.zendesk.com/components/select).

Reply:
The `ColorDialog` has an API the covers the design specs. Are you suggesting an API that allows consumers to customize the dialog trigger?

Reply:
> How can I invoke the ColorDialog with an alternate trigger (i.e. IconButton)?

Thanks for bringing this up because this was a missing feature. The API has been modified to accept any (valid) element via `props.children` as a trigger. I've created a Storybook example to demonstrate the usage with an `IconButton`.